ocaml/xapi/repository.ml

```diff
@@ -543,7 +543,24 @@ let get_repository_handler (req : Http.Request.t) s _ =
         match String.sub_to_end req.Request.uri len with
         | uri_path ->
             let root = !Xapi_globs.local_pool_repo_dir in
-            Fileserver.response_file s (Helpers.resolve_uri_path ~root ~uri_path)
+            let uri_path' =
+              match
+                String.starts_with ~prefix:Constants.enabled_repo_uri_prefix
```

Comment:
More strict match would filter out a case like `/repository/enabledxxx` or `/repository/enabled/yyy`

Reply:
I don't know how this is usually handled but should we split the paths into components and match on `"repository" :: "enabled" :: _`. I would prefer that over string prefix. We should try to use a higher semantic level than pure strings.